src/main/java/cubicchunks/CommonEventHandler.java

```diff
@@ -133,11 +134,11 @@ public void onPlayerJoinWorld(EntityJoinWorldEvent evt) {
     public static class ClientProxy extends ServerProxy {
 
         // shouldSkipWorld
-        @Override public boolean test(ICubicWorld world) {
-            if (super.test(world)) {
-                return true;
-            }
-            if (((World) world).getChunkProvider().getClass() != ChunkProviderClient.class) {
+        @Override
+        public boolean test(ICubicWorld world) {
+            if (!super.test(world))
+                return false;
+            if (world.getClass() == (Class<?>) WorldClient.class && ((World) world).getChunkProvider().getClass() != ChunkProviderClient.class) {
```

Comment:
i think we should just have a hardcoded list of allowed classes in each proxy, and we would just check if the list contains the class. Much less error-prone.

It would need list of allowed:
 * World classes implementations
 * Chunk privider implementations
 * anything more?